controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -184,6 +177,23 @@ protected void startUp() {
                         .thenAcceptAsync(x -> controllerEventProcessors.startAsync(), controllerExecutor);
             }
 
+            // Setup and start controller cluster listener after all sweepers have been initialized.
+            if (serviceConfig.isControllerClusterListenerEnabled()) {
```

Comment:
I can't really assess the impact of moving this code here. Is this going to create any problem, like a cyclic dependency such as the one we had before? I'm worried that this code wasn't here before for some reason.